src/app/server/Mdns.cpp

```diff
@@ -78,37 +78,204 @@ chip::ByteSpan FillMAC(uint8_t (&mac)[8])
 
 } // namespace
 
-uint16_t gSecuredPort   = CHIP_PORT;
-uint16_t gUnsecuredPort = CHIP_UDC_PORT;
+#if CHIP_DEVICE_CONFIG_ENABLE_EXTENDED_DISCOVERY
+
+constexpr const char kExtendedDiscoveryTimeoutKeypairStorage[] = "ExtDiscKey";
+
+void MdnsServer::SetExtendedDiscoveryTimeoutSecs(int16_t secs)
+{
+    ChipLogDetail(Discovery, "SetExtendedDiscoveryTimeoutSecs %d", secs);
+    chip::DeviceLayer::PersistedStorage::KeyValueStoreMgr().Put(kExtendedDiscoveryTimeoutKeypairStorage, &secs, sizeof(secs));
+}
+
+int16_t MdnsServer::GetExtendedDiscoveryTimeoutSecs()
+{
+    CHIP_ERROR err = CHIP_NO_ERROR;
+    int16_t secs;
+
+    err = chip::DeviceLayer::PersistedStorage::KeyValueStoreMgr().Get(kExtendedDiscoveryTimeoutKeypairStorage, &secs, sizeof(secs));
+    if (err != CHIP_NO_ERROR)
+    {
+        ChipLogError(Discovery, "Failed to get extended timeout configuration err: %s", chip::ErrorStr(err));
+        secs = CHIP_DEVICE_CONFIG_EXTENDED_DISCOVERY_TIMEOUT_SECS;
+    }
+
+    ChipLogDetail(Discovery, "GetExtendedDiscoveryTimeoutSecs %d", secs);
+    return secs;
+}
+
+/// Callback from Extended Discovery Expiration timer
+void HandleExtendedDiscoveryExpiration(System::Layer * aSystemLayer, void * aAppState)
+{
+    MdnsServer::Instance().OnExtendedDiscoveryExpiration(aSystemLayer, aAppState);
+}
+
+/// Checks if extended discovery has expired and if so,
+/// stops commissionable node advertising
+/// Extended Discovery Expiration refers here to commissionable node advertising when NOT in commissioning mode
+void MdnsServer::OnExtendedDiscoveryExpiration(System::Layer * aSystemLayer, void * aAppState)
+{
+    if (mExtendedDiscoveryExpirationMs == TIMEOUT_CLEARED)
```

Comment:
This seems to share a lot of code in common with the 'OnDiscoveryExpiration' below.
Could you share the code?

Reply:
created a shared method to call